src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -1878,6 +1903,10 @@ public override BoundNode VisitParameter(BoundParameter node)
             {
                 CheckAssigned(node.ParameterSymbol, node.Syntax);
             }
+            else if (CurrentSymbol is SynthesizedRecordConstructor)
+            {
+                NoteRead(node.ParameterSymbol);
```

Comment:
>node.ParameterSymbol [](start = 25, length = 20)

How do we know parameter and CurrentSymbol are related?

Reply:
In lambda scenarios, CurrentSymbol is the lambda, not the primary constructor.
I'll add tests, but I believe that's the behavior we want (ie. reading the record parameter in a lambda doesn't count).

---
In reply to: [512809160](https://github.com/dotnet/roslyn/pull/48895#discussion_r512809160) [](ancestors = 512809160)

Reply:
> but I believe that's the behavior we want (ie. reading the record parameter in a lambda doesn't count).

I don't know why would this be the behavior we want. Nevertheless, it would be good to clearly specify in the PR description what we are trying to achieve in this PR.  

---
In reply to: [512902651](https://github.com/dotnet/roslyn/pull/48895#discussion_r512902651) [](ancestors = 512902651,512809160)

Reply:
Thinking more about this, maybe the usage inside the lambda should count as well. Made the change. Thanks

---
In reply to: [512907636](https://github.com/dotnet/roslyn/pull/48895#discussion_r512907636) [](ancestors = 512907636,512902651,512809160)